File: backend/src/main/java/com/anmory/yunji/entity/CheckReport.java
package com.anmory.yunji.entity;

import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
public class CheckReport {
    private Integer reportId;
    private Integer userId;
    private String fileUrl;
    private String originalFilename;
    private String parsedSummary;
    private LocalDate nextCheckDate;
    private Boolean emailSent;
    private LocalDateTime lastSendAt;
    private String sendStatus;
    private Integer retryCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/lib/api/check-report.ts
import { apiGet, apiUpload } from "@/lib/api"

export interface CheckReportItem {
  reportId: number
  userId: number
  fileUrl: string
  originalFilename?: string
  parsedSummary?: string
  nextCheckDate?: string
  emailSent?: boolean
  sendStatus?: string
  retryCount?: number
  createdAt?: string
  updatedAt?: string
}

export async function uploadCheckReport(userId: number, file: File, extraText?: string) {
  return apiUpload<CheckReportItem>("/api/check-report/upload", { key: "file", file }, { userId, extraText })
}

export async function listCheckReports(userId: number) {
  return apiGet<CheckReportItem[]>("/api/check-report/list", { userId })
}

